### ISSUE TITLE
undefined constant error for association defition

### ISSUE THREAD

**Issue by sdzyba** [luckyframework/lucky_record#265]
Crystal version: 0.25.1
LuckyRecord version: 0.6.0

When two models `A` and `B` are defined in the project and model `A` is associated to `B` like so:
```
table :a do
  ...
  belongs_to b : B
end
```
user would get a `undefined constant B` error.

My guess is that model files are read in alphabetical order and while parsing model `A` the required model `B` is not loaded yet.

Workaround: add `require "./b"` at the top of model `A` definition file.

**Comment by paulcsmith** [comment]
@sdzyba You are correct about the alphabetical ordering and adding the require statement. This is something in Crystal more than in LuckyRecord. There is an issue open in the Crystal repo for showing a helpful error. I may also add something specific to LuckyRecord to help people in this specific case.

I don't think there's a way around this so giving more helpful errors is probably the best bet

**Comment by edwardloveall** [comment]
I just thought of a possible solution to this, what if we do a pass to define an empty constant for each model before the full implementation is defined? Here's a proof of concept:

```crystal
# Initial empty "model" defined
class User
end

# Model referencing other model
class Admin < User
end

# Actual model definition
class User
  def magic
    "magic"
  end
end

Admin.new.magic #=> "magic"
```

This might defeat some of the type safety in a way I didn't think of, in which case we shouldn't do it. Just throwing it out there 🙂

**Comment by edwardloveall** [comment]
Mmm, I just tried this and it doesn't work so well. At least I had trouble getting _enough_ of a `User` to show up to make the `Admin` not throw an error when it referenced the `User`
